### PR TITLE
add outbound mark to tproxy.

### PR DESCRIPTION
**Pull request by springzfx**
## add outbound mark to tproxy.
set to 0, means won't change anything
default set to 255.
Screenshot_20200423_120557

### CODE REVIEW

**Comment by huyizheng** [comment]
另外现在这个版本，设置mark时会把复杂配置里的mark也覆盖掉

**Comment by springzfx** [comment]
> 另外现在这个版本，设置mark时会把复杂配置里的mark也覆盖掉
那就只针对非复杂设置的吧

File: makespec/BUILDVERSION [hunk 1]
```diff
@@ -1 +1,5 @@
-5281
\ No newline at end of file
+<<<<<<< HEAD
+5305
+=======
+5281
+>>>>>>> 91eb8ee1cb0a5d5e1dc9a838cd6f5b90a483e2b5
```
Comment (Qv2ray-dev) [inline makespec/BUILDVERSION:5]: What's this?

File: src/core/connection/Generation.cpp [hunk 2]
```diff
@@ -555,7 +557,46 @@ namespace Qv2ray::core::connection
                 //
                 root["api"] = GenerateAPIEntry(API_TAG_DEFAULT);
             }
+
+            if (!isComplex && GlobalConfig.inboundConfig.useTPROXY && GlobalConfig.outboundConfig.mark > 0)
```
Comment (Qv2ray-dev) [inline src/core/connection/Generation.cpp:561]: I suggest moving this check into the `simple connection config` if-branch, for simplicity.

**Comment by springzfx** [comment]
LGTM now.

File: makespec/BUILDVERSION (hunk 1 above)
Comment (springzfx) [reply makespec/BUILDVERSION:5]: I don't known anything about this. automaticly?

File: src/core/connection/Generation.cpp (hunk 2 above)
Comment (Qv2ray-dev) [reply src/core/connection/Generation.cpp:561]: anyway, it just works.

**Pull request merged**